ms-engine-spring-jaxrs/src/main/java/net/trajano/ms/engine/jaxrs/JaxRsRouter.java

```diff
@@ -0,0 +1,103 @@
+package net.trajano.ms.engine.jaxrs;
+
+import static java.util.Arrays.stream;
+
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.ws.rs.ApplicationPath;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Application;
+
+import org.reflections.Reflections;
+
+import io.vertx.core.Handler;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * This will take the JAX-RS annotated classes and create individual routes on a
+ * VertX router.
+ */
+public class JaxRsRouter {
+
+    public void register(final Class<? extends Application> applicationClass,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 35 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)